tasks/yesno.py

```diff
@@ -0,0 +1,301 @@
+        c['Ddim'] = .1
+        c['loss'] = 'binary_crossentropy'
+        c['nb_epoch'] = 100
+        c['batch_size'] = 10
+        c['class_mode'] = 'binary'
+
+        # old rnn
+        c['pdim'] = 2.5
+        c['pact'] = 'tanh'
+
+    def load_set(self, fname, cache_dir=None):
+        # TODO: Make the cache-handling generic,
+        # and offer a way to actually pass cache_dir
+        save_cache = False
+        if cache_dir:
+            import os.path
+            fname_abs = os.path.abspath(fname)
+            from hashlib import md5
+            cache_filename = "%s/%s.p" % (cache_dir, md5(fname_abs.encode("utf-8")).hexdigest())
+            try:
+                with open(cache_filename, "rb") as f:
+                    return pickle.load(f)
+            except (IOError, TypeError, KeyError):
+                save_cache = True
+
+        s0, s1, y, kw, akw, t = loader.load_anssel(fname, skip_oneclass=False)
+
+        if self.vocab is None:
+            vocab = Vocabulary(s0 + s1)  # FIXME: lower?
+        else:
+            vocab = self.vocab
+
+        si0 = vocab.vectorize(s0, spad=self.c['spad'])
+        si1 = vocab.vectorize(s1, spad=self.c['spad'])
+        f0, f1 = nlp.sentence_flags(s0, s1, self.c['spad'], self.c['spad'])
+        gr = graph_input_anssel(si0, si1, y, f0, f1, s0, s1, kw=kw, akw=akw)
+        gr, y = self.merge_questions(gr)
+        if save_cache:
+            with open(cache_filename, "wb") as f:
+                pickle.dump((s0, s1, y, vocab, gr), f)
+                print("save")
+
+        return (gr, y, vocab)
+
+    def build_model(self, module_prep_model, do_compile=True):
+
+        model = build_model(self.emb, self.vocab, module_prep_model, self.c)
+
+        # for lname in self.c['fix_layers']:  # whats this??
+        #     model.nodes[lname].trainable = False
+
+        if do_compile:
+            model.compile(loss={'score': self.c['loss']}, optimizer=self.c['opt'])
+        return model
+
+    def fit_callbacks(self, weightsf):
+        return [ModelCheckpoint(weightsf, save_best_only=True, monitor='val_loss', mode='min'),
+                EarlyStopping(monitor='val_loss', mode='min', patience=10)]
+
+    def eval(self, model):
+        res = []
+        for gr, fname in [(self.gr, self.trainf), (self.grv, self.valf), (self.grt, self.testf)]:
+            if gr is None:
+                res.append(None)
+                continue
+            loss, acc = model.evaluate(gr, show_accuracy=True)
+            res.append(YesNoRes(loss, acc))
+        return tuple(res)
+
+    def res_columns(self, mres, pfx=' '):
+        """ Produce README-format markdown table row piece summarizing
+        important statistics """
+        return('%s%.6f    |%s%.6f |%s%.6f |%s%.6f'
+               % (pfx, mres[self.trainf]['Loss'],
+                  pfx, mres[self.valf]['Precision'],
+                  pfx, mres[self.testf].get('Loss', np.nan),
+                  pfx, mres[self.testf].get('Precision', np.nan)))
```

Comment:
Interesting choice.  Why train + test reports loss (what does it tell us compared to precision?) while val reports precision?